lgc/patch/PatchResourceCollect.cpp

```diff
@@ -1375,15 +1377,49 @@ void PatchResourceCollect::matchGenericInOut() {
     LLPC_OUTS("\n");
   }
 
+  if (!perPrimitiveInLocMap.empty()) {
+    assert(inOutUsage.perPrimitiveInputMapLocCount == 0);
+    for (auto locMap : perPrimitiveInLocMap) {
+      assert(m_shaderStage == ShaderStageFragment && locMap.second != InvalidValue);
+      inOutUsage.perPrimitiveInputMapLocCount = std::max(inOutUsage.perPrimitiveInputMapLocCount, locMap.second + 1);
+      LLPC_OUTS("(" << getShaderStageAbbreviation(m_shaderStage) << ") Input (per-primitive):  loc = " << locMap.first
+                    << "  =>  Mapped = " << locMap.second << "\n");
+    }
+    LLPC_OUTS("\n");
+  }
+
+  if (!perPrimitiveOutLocMap.empty()) {
+    assert(inOutUsage.perPrimitiveOutputMapLocCount == 0);
+    for (auto locMap : perPrimitiveOutLocMap) {
+      assert(m_shaderStage == ShaderStageMesh && locMap.second != InvalidValue);
+      inOutUsage.perPrimitiveOutputMapLocCount = std::max(inOutUsage.perPrimitiveOutputMapLocCount, locMap.second + 1);
+      LLPC_OUTS("(" << getShaderStageAbbreviation(m_shaderStage) << ") Output (per-primitive): loc = " << locMap.first
+                    << "  =>  Mapped = " << locMap.second << "\n");
+    }
+    LLPC_OUTS("\n");
+  }
```

Comment:
Is there a case where `inOutUsage.perPrimitive*MapLocCount != 0` before we reach this code? Asked differently, could the outer `empty` check just be removed in both cases?

Reply:
Yes, you are right. The outer if empty check is redundant. Will remove it.